elasticdl/doc/async_sgd_design.md

```diff
@@ -0,0 +1,73 @@
+# Design Doc: Asynchronous SGD (Stochastic Gradient Descent)
+
+## Motivation
```

Comment:
This Motivation section is more about @chunyang-wen 's personal understanding of the field and the project. The motivation for us to add asynchronous SGD is because the synchronous update runs slow.